src/3d/qgs3dmapsettings.cpp

```diff
@@ -411,12 +415,12 @@ void Qgs3DMapSettings::resolveReferences( const QgsProject &project )
 
 QgsVector3D Qgs3DMapSettings::mapToWorldCoordinates( const QgsVector3D &mapCoords ) const
 {
-  return Qgs3DUtils::mapToWorldCoordinates( mapCoords, mOrigin );
+  return Qgs3DUtils::mapToWorldCoordinates( mapCoords, mOrigin ) + QgsVector3D( 0.0f, mTerrainElevationOffset, 0.0f );
```

Comment:
Hmm I am not sure if this is correct... I think map-world coordinates should still have the conversion unaffected by the vertical offset for terrain...

Reply:
That depends on whether we want the world coordinates in this context to be the real world coordinates in the opengl scene or disregarding the terrain elevation. I think the elevation offset should be included Qgs3DUtils::mapToWorldCoordinates but I'm not sure how to do that

Reply:
After some thoughts, I think you are right. The terrain elevation offset should only influence the terrain not apply a translation on the world orign. I reverted baack these modifications